File: go/src/MaximumDepthofBinaryTree/104.go
package MaximumDepthofBinaryTree

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func maxDepth(root *TreeNode) int {
	if root == nil {
		return 0
	}

	ltree := maxDepth(root.Left)
	rtree := maxDepth(root.Right)

	if ltree > rtree {
		return ltree + 1
	} else {
		return rtree + 1
	}
}

File: python3/MaximumDepthofBinaryTree/104.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def maxDepth(self, root: TreeNode) -> int:
        if root is None:
            return 0
        
        ltree = self.maxDepth(root.left)
        rtree = self.maxDepth(root.right)
        
        if ltree > rtree:
            return ltree+1
        else:
            return rtree+1
        